### ISSUE TITLE
Weird behavior

### ISSUE THREAD

**Issue by Themandunord** [javascript-obfuscator/javascript-obfuscator#119]
Hi @sanex3339 :)

I have a weird behavior with your lib, and I can't handle it...

I have a function with a Promise that return this : 
`resolve({response: response, suggestions: {type: 'STARTSUGGESTION', items: items}});`

In other function, I call this function and I have : 
`return this.generateThemeSuggestion(response, locale).then(({response, suggestions}) =>`

Your lib transform this code to this : 
`_0x5e6f9a({'response': _0x2639a8, 'suggestions': {'type': _0xdaa6('0x80'), 'items': _0x110d2c }});`

`return this[_0xdaa6('0x8')](_0x20d6fc, _0x328b92)[_0xdaa6('0x7')](({_0x20d6fc, suggestions}) =>`

So the problem is that js can't extract the property 'response' because the promise export `response : _0x2639a8` and the then try to extract the `_0x2639a8`...

Do you see the problem ? And do you see a solution ?

(This problem is very urgent and I need that for the end of the week :( )

Thanks you :)

**Comment by borzovplus** [comment]
+ 1 same problem !

**Comment by sanex3339** [comment]
I'll fix it soon

**Comment by sanex3339** [comment]
what version you using?

**Comment by sanex3339** [comment]
Can't reproduce with code like:
```ts
(function(){
    function Foo (response) {
        return new Promise((resolve) => {
            resolve({response: response, suggestions: 1});
        });
    }
    
    function Bar () {
        Foo(123).then(({response, suggestions}) => console.log(response));
    }
    
    Bar();
})();
```

Can you provide slice of the code that will trigger this error?

**Comment by sanex3339** [comment]
Code to reproduce the error:

```ts
(function(){
    function foo (data) {
        new Promise((resolve) => resolve({data: data}))
            .then(({data}) => console.log(data));
    }
    
    foo(1);
})();
```

**Comment by sanex3339** [comment]
https://github.com/javascript-obfuscator/javascript-obfuscator/pull/120

**Comment by sanex3339** [comment]
I'll release 0.12.2 version later today or tomorrow

**Comment by sanex3339** [comment]
Fixed in `javascript-obfuscator@0.12.2`
